src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
- *
- * @author emil
- */
+import de.mediathekview.mlib.Config;
+import de.mediathekview.mlib.daten.DatenFilm;
+import de.mediathekview.mlib.daten.ListeFilme;
+import de.mediathekview.mlib.tool.Hash;
+import de.mediathekview.mlib.tool.Log;
+import de.mediathekview.mlib.tool.MVHttpClient;
+import mServer.tool.MserverDaten;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
 public class AddToFilmlist {
-    private static final EtmMonitor etmMonitor = EtmManager.getEtmMonitor();
-    AtomicInteger threadCounter = new AtomicInteger(0);
-    AtomicInteger treffer = new AtomicInteger(0);
-    ListeFilme vonListe;
-    ListeFilme listeEinsortieren;
-    Collection<DatenFilm> filteredOnline = new ArrayList<>();
+    /**
+     * Minimum size of films in MiB to be included in new list.
+     */
+    private static final int MIN_SIZE_ADD_OLD = 5;
+    private final static int NUMBER_OF_THREADS = 32;//(Runtime.getRuntime().availableProcessors() * Runtime.getRuntime().availableProcessors()) / 2;
+    private final ListeFilme vonListe;
+    private final ListeFilme listeEinsortieren;
+    /**
+     * List of all locally started import threads.
+     */
+    private final ArrayList<ImportOldFilmlistThread> threadList = new ArrayList<>();
+    private AtomicInteger threadCounter = new AtomicInteger(0);
+    
 
     public AddToFilmlist(ListeFilme vonListe, ListeFilme listeEinsortieren) {
         this.vonListe = vonListe;
         this.listeEinsortieren = listeEinsortieren;
     }
 
     public synchronized void addLiveStream() {
-        EtmPoint performancePoint = etmMonitor.createPoint("AddToFilmlist:addLiveStream");
-        // live-streams einfügen, es werde die vorhandenen ersetzt!
-
-        if (listeEinsortieren.size() <= 0) {
-            //dann wars wohl nix
-            return;
-        }
+        if (listeEinsortieren.size() <= 0) return;
 
-        Iterator<DatenFilm> it = vonListe.iterator();
-        while (it.hasNext()) {
-            DatenFilm f = it.next();
-            if (f.arr[DatenFilm.FILM_THEMA].equals(ListeFilme.THEMA_LIVE)) {
-                it.remove();
-            }
-        }
+        vonListe.removeIf(f -> f.arr[DatenFilm.FILM_THEMA].equals(ListeFilme.THEMA_LIVE));
         listeEinsortieren.forEach(vonListe::add);
-        performancePoint.collect();
     }
 
-    public synchronized int addOldList() {
-        // in eine vorhandene Liste soll eine andere Filmliste einsortiert werden
-        // es werden nur Filme die noch nicht vorhanden sind, einsortiert
-        threadCounter = new AtomicInteger(0);
-        treffer = new AtomicInteger(0);
-        int size = listeEinsortieren.size();
-        HashSet<String> hash = new HashSet<>(listeEinsortieren.size() + 1, 1);
-
-        Duration.staticPing("AddOld-1");
+    private void performTitleSearch(HashSet<Hash> hash, final int size) {
+        vonListe.parallelStream().forEach(f -> {
+            synchronized (hash) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') "hash" is a method parameter, and should not be used for synchronization. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2445)